platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/DefaultBuildTreeModelControllerServices.kt

```diff
@@ -238,14 +240,24 @@ class DefaultBuildTreeModelControllerServices : BuildTreeModelControllerServices
         // This was originally only for the configuration cache, but now used for configuration cache and problems reporting
         registration.add(ProblemFactory::class.java, DefaultProblemFactory::class.java)
 
+        registration.add(ConfigurationCacheProblemsListener::class.java, DefaultConfigurationCacheProblemsListener::class.java)
+        // Set up CC problem reporting pipeline, based on the build configuration
+        when {
+            modelParameters.isConfigurationCache -> registration.add(ConfigurationCacheProblems::class.java)
+            allowNudgingToEnableConfigurationCache(requirements, modelParameters) -> {
+                // Don't suggest enabling CC if it is on, even if implicitly (e.g. enabled by isolated projects).
```

Comment:
nitpicking, but do we need to block here? This helpful comment could go before the alternative (between lines 246 and 247), or between the arrow and `registration.add(ConfigurationCachePromoHandler::class.java)`?

Reply:
> 💅
> 
> nitpicking, but do we need to block here? This helpful comment could go before the alternative (between lines 246 and 247), or between the arrow and `registration.add(ConfigurationCachePromoHandler::class.java)`?

Looking closely into this, I realized that the comment doesn't make much sense in its current place. I've changed the structure a bit.